bot/exts/events/advent_of_code/_cog.py

```diff
@@ -52,13 +60,80 @@ def __init__(self, bot: Bot):
         self.status_task.set_name("AoC Status Countdown")
         self.status_task.add_done_callback(_helpers.background_task_callback)
 
+        self.completer_task.start()
+
+    @tasks.loop(minutes=10.0)
+    async def completer_task(self) -> None:
+        """
+        Synchronise the members with the completionist role and members who has completed all 50 AoC stars.
+
+        Adds the role to those that need it, removes the role from those who have been added ot the block list.
+        Runs on a schedule, as defined in the task.loop decorator.
+        """
+        await self.bot.wait_until_guild_available()
+        guild = self.bot.get_guild(Client.guild)
+        completionist_role = guild.get_role(Roles.aoc_completionist)
```

Comment:
Couldn't hurt to add a check that this succeeds in getting the role. I had a misconfiguration when testing locally and it resulted in a traceback that was confusing at first glance.

Reply:
Done in https://github.com/python-discord/sir-lancebot/pull/991/commits/2fe89228fd3e583f2c8746d7df08af9c1c234972